src/server/req_quejob.c

```diff
@@ -671,12 +671,16 @@ req_quejob(struct batch_request *preq)
 
 		/* decode attribute */
 #ifndef PBS_MOM
-		presv = find_resv_by_quename(qname);
+		if (index == JOB_ATR_create_resv_from_job) {
+			if (qname != NULL && *qname != '\0') {
+				presv = find_resv_by_quename(qname);
```

Comment:
If this is the only time you use presv, move the declaration to this block.  Also, you added a initialization to NULL, I'm not sure you need it.